actions-cleanup-openstack-vms/action.yaml

```
@@ -64,24 +64,32 @@ runs:
         echo "OS_PROJECT_DOMAIN_ID=${{ inputs.PROJECT_DOMAIN_ID }}" >> $GITHUB_ENV
         echo "OS_USER_DOMAIN_ID=${{ inputs.USER_DOMAIN_ID }}" >> $GITHUB_ENV
         echo "OS_IDENTITY_API_VERSION=3" >> $GITHUB_ENV
+    - name: create artifacts folder
+      id: temp-dir
+      shell: bash
+      run: |
+        TEMP=$(mktemp -d)
+        echo "Temp directory: ${TEMP}"
+        echo "TEMP=$TEMP" >> $GITHUB_OUTPUT
     - name: Get list of VMs
       shell: bash
-      run: openstack server list -f value -c ID
+      run: echo "VM_LIST=$(openstack server list -f value -c ID)" >> $GITHUB_ENV
       id: vm_list
     - name: Delete old VMs
       shell: bash
       run: |-
         set -e
-        mkdir -p ../vm_artifacts
         delta=$(date -d "${{ inputs.AGE_HOURS }} hours ago" +%s)
-        if [ -z "{{ steps.vm_list.outputs.stdout }}" ]; then
-          for vm in ${{ steps.vm_list.outputs.stdout }}; do
+        if [ -n "$VM_LIST" ]; then
+          for vm in "$VM_LIST"; do
```

Are you sure it works ? 
To me, It will not work as you expect, since you quote `VM_LIST` it's a string so here you will basically have the full list in the first `vm` variable

```suggestion
          for vm in $VM_LIST; do
```